service/src/main/java/uk/nhs/adaptors/oneoneone/cda/report/mapper/CompositionMapper.java

```diff
@@ -110,11 +124,48 @@ public Composition mapComposition(POCDMT000002UK01ClinicalDocument1 clinicalDocu
             }
         }
 
+        //TODO - add list of careplans into sections above, update tests also
+        List<DomainResource> resources = new ArrayList<>(carePlans);
+        for (Reference reference : encounter.getIncomingReferral()) {
```

Comment:
Could be changed to:
```
encounter.getIncomingReferral().stream()
            .filter(reference -> reference.getResource().fhirType().equals(ResourceType.ReferralRequest.name()))
            .map(DomainResource.class::cast)
            .forEach(resources::add);
```

Reply:
The code you provided needs a slight modification in order to work, is this fine?

```
encounter.getIncomingReferral().stream()
            .filter(reference -> reference.getResource().fhirType().equals(ResourceType.ReferralRequest.name()))
            .map(reference -> (DomainResource) reference.getResource())
            .forEach(resources::add);
```